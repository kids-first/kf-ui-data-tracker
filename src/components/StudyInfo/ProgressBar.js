import React from 'react';
import PropTypes from 'prop-types';
import {Progress} from 'semantic-ui-react';
import {steppingFields} from '../../common/notificationUtils';
/**
 * Displays progress bar component indication the completion of each info step
 * - Show as red when none of the fields in current step have values
 * - Show as yellow when one or more fields have values but not completed yet
 * - Show as green when all the field are given values.
 */
const ProgressBar = ({values, step}) => {
  var filled = 0;
  steppingFields[step].map(field => {
    if (values[field] && (values[field] > 0 || values[field].length > 0)) {
      filled += 1;
    }
    return true;
  });
  const total = steppingFields[step].length;
  return (
    <Progress
      className="progress-bar--custom"
      value={filled}
      total={total}
      size="tiny"
      label={filled + '/' + total + ' fields complete'}
      error={filled === 0}
      warning={filled > 0 && filled < total}
      success={filled === total}
    />
  );
};

ProgressBar.propTypes = {
  /** The value object from Formik */
  values: PropTypes.object,
  /** The step name, each step has its own array of fields to track */
  step: PropTypes.number,
};

export default ProgressBar;
