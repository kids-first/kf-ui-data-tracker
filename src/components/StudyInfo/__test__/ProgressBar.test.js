import React from 'react';
import {render} from 'react-testing-library';
import ProgressBar from '../ProgressBar';

it('renders progress bar with 0/3 completed', () => {
  const values = {
    bucket: '',
    name: '',
    shortName: '',
  };
  const tree = render(<ProgressBar values={values} step={0} />);
  expect(tree.container).toMatchSnapshot();
});

it('renders progress bar with 2/3 completed', () => {
  const values = {
    bucket: '',
    name: 'incubate dynamic content',
    shortName: 'shortName',
  };
  const tree = render(<ProgressBar values={values} step={0} />);
  expect(tree.container).toMatchSnapshot();
});

it('renders progress bar with 3/3 completed', () => {
  const values = {
    bucket: 'authority-result',
    name: 'incubate dynamic content',
    shortName: 'shortName',
  };
  const tree = render(<ProgressBar values={values} step={0} />);
  expect(tree.container).toMatchSnapshot();
});

it('renders progress bar in (no data) error stage', () => {
  const tree = render(<ProgressBar />);
  expect(tree.container).toMatchSnapshot();
});

it('renders progress bar in (invalid data) error stage', () => {
  const tree = render(<ProgressBar step={4} />);
  expect(tree.container).toMatchSnapshot();
});
