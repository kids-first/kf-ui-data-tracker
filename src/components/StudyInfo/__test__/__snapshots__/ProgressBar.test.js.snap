// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders progress bar in (invalid data) error stage 1`] = `
<div>
  <div
    class="ui tiny error success progress progress-bar--custom"
    data-percent="0"
  >
    <div
      class="bar"
      style="width: 0%;"
    />
    <div
      class="label"
    >
      Invalid  progress
    </div>
  </div>
</div>
`;

exports[`renders progress bar in (no data) error stage 1`] = `
<div>
  <div
    class="ui tiny error success progress progress-bar--custom"
    data-percent="0"
  >
    <div
      class="bar"
      style="width: 0%;"
    />
    <div
      class="label"
    >
      Invalid  progress
    </div>
  </div>
</div>
`;

exports[`renders progress bar with 0/3 completed 1`] = `
<div>
  <div
    class="ui tiny error progress progress-bar--custom"
    data-percent="0"
  >
    <div
      class="bar"
      style="width: 0%;"
    />
    <div
      class="label"
    >
      0/3 fields complete
    </div>
  </div>
</div>
`;

exports[`renders progress bar with 2/3 completed 1`] = `
<div>
  <div
    class="ui tiny warning progress progress-bar--custom"
    data-percent="66"
  >
    <div
      class="bar"
      style="width: 66.66666666666666%;"
    />
    <div
      class="label"
    >
      2/3 fields complete
    </div>
  </div>
</div>
`;

exports[`renders progress bar with 3/3 completed 1`] = `
<div>
  <div
    class="ui tiny success progress progress-bar--custom"
    data-percent="100"
  >
    <div
      class="bar"
      style="width: 100%;"
    />
    <div
      class="label"
    >
      3/3 fields complete
    </div>
  </div>
</div>
`;
